src/java.base/share/classes/java/lang/StringLatin1.java

```diff
@@ -46,7 +44,7 @@
 
     public static char charAt(byte[] value, int index) {
         checkIndex(index, value.length);
-        return (char)(value[index] & 0xff);
+        return getChar(value, index);
```

Comment:
Seems unrelated to the gist of the patch, and again might skew startup profiles.

Reply:
Reverted